contracts/contracts/vault/OETHVaultAdmin.sol

```diff
@@ -1,12 +1,113 @@
+
+    constructor(address _weth) {
+        weth = _weth;
+    }
+
+    /// @dev Simplified version of the deposit function as WETH is the only supported asset.
+    function _depositToStrategy(
+        address _strategyToAddress,
+        address[] calldata _assets,
+        uint256[] calldata _amounts
+    ) internal override {
+        require(
+            strategies[_strategyToAddress].isSupported,
+            "Invalid to Strategy"
+        );
+        require(
+            _assets.length == 1 && _amounts.length == 1 && _assets[0] == weth,
+            "Only WETH is supported"
+        );
+
+        // Check the there is enough WETH to transfer once the WETH reserved for the withdrawal queue is accounted for
+        require(_amounts[0] <= _wethAvailable(), "Not enough WETH available");
+
+        // Send required amount of funds to the strategy
+        IERC20(weth).safeTransfer(_strategyToAddress, _amounts[0]);
+
+        // Deposit all the funds that have been sent to the strategy
+        IStrategy(_strategyToAddress).depositAll();
+    }
+
+    function _withdrawFromStrategy(
+        address _recipient,
+        address _strategyFromAddress,
+        address[] calldata _assets,
+        uint256[] calldata _amounts
+    ) internal override {
+        super._withdrawFromStrategy(
+            _recipient,
+            _strategyFromAddress,
+            _assets,
+            _amounts
+        );
+
+        IVault(address(this)).addWithdrawalQueueLiquidity();
+    }
+
+    function _withdrawAllFromStrategy(address _strategyAddr) internal override {
+        super._withdrawAllFromStrategy(_strategyAddr);
+
+        IVault(address(this)).addWithdrawalQueueLiquidity();
+    }
+
+    function _withdrawAllFromStrategies() internal override {
+        super._withdrawAllFromStrategies();
+
+        IVault(address(this)).addWithdrawalQueueLiquidity();
+    }
+
+    /// @dev Calculates the amount of WETH in the Vault that is not reserved for the withdrawal queue.
+    function _wethAvailable() internal view returns (uint256 wethAvailable) {
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+
+        // The amount of WETH that is still to be claimed in the withdrawal queue
+        uint256 outstandingWithdrawals = queue.queued - queue.claimed;
+
+        // The amount of sitting in WETH in the vault
+        uint256 wethBalance = IERC20(weth).balanceOf(address(this));
+
+        // If there is more WETH in the vault than the outstanding withdrawals
+        if (wethBalance > outstandingWithdrawals) {
+            wethAvailable = wethBalance - outstandingWithdrawals;
+        }
+    }
+
+    function _swapCollateral(
+        address _fromAsset,
+        address _toAsset,
+        uint256 _fromAssetAmount,
+        uint256 _minToAssetAmount,
+        bytes calldata _data
+    ) internal override returns (uint256 toAssetAmount) {
+        require(_fromAsset != weth, "Swap from WETH not supported");
```

Comment:
If the vault only has WETH, and no other collaterals, can we just make this revert?

Reply:
I've made this change